quantumblacklabs/kedro-viz
# [KED-2381] Refactor zoom and views

## Description

Abstracts and improves the viewport implementation. 
This follows up recent view work from #356.

## Development notes

- moves viewport code into a separate `view` utility
- this simplifies and unifies the flowchart and minimap components
- provides a more consistent and convenient API for this usage
- improves test coverage throughout

## QA notes

Behaviour should not change.

All flowchart and minimap functionality involving view effects should be tested.

Refer to notes in #356 for rules on how the view should now act.

This will include testing in various combinations:

- open / closed sidebar and metadata panel
- resizing the browser to various viewport sizes / aspect ratios
- different graph aspect ratios (square, wide, tall)
- selecting elements on the sidebar and on the chart
- selecting tags and pipelines
- toggling labels
- using the minimap

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
